packages/cli/src/command_middleware.test.ts

```diff
@@ -40,6 +40,7 @@ void describe('commandMiddleware', () => {
         process.env.AWS_ACCESS_KEY_ID = testAccessKeyId;
         process.env.AWS_SECRET_ACCESS_KEY = testSecretAccessKey;
         process.env.AWS_REGION = testRegion;
+        delete process.env.AWS_PROFILE;
       });
 
       afterEach(() => {
```

Comment:
This test should capture whatever the initial value of all of these env vars is and reset them at the end of the suite. We shouldn't be relying on them in unit tests, but better to minimize the chance of side effects

Reply:
this makes me think that perhaps we should
`new CommandMidleWare(..., process.env)` 
I.e. inject env into ctor and eliminate the problem altogether.

Reply:
unfortunately we can't. Our code set `process.env.AWS_PROFILE` but aws lib like `loadConfig` and `fromNodeProviderChain` read other envVars directly from process.env
I updated the code to restore all used envVars. Pls take a look.